llvm/tools/llvm-no-spir-kernel/llvm-no-spir-kernel.cpp

```diff
@@ -6,15 +6,22 @@
 //
 //===----------------------------------------------------------------------===//
 //
-// This utility checks if the input module contains functions that is a spir
-// kernel. Return 0 if no, return 1 if yes. Use of an output file is not
-// required for a successful check. It is used to allow for proper input and
-// output flow within the driver toolchain.
+// This utility checks if the input module contains functions that are a SPIR
+// kernel.
+//
+// - Return 0 if the LLVM module is "clean" from SPIR kernels
+// - Return 1 upon the first SPIR kernel occurence
+// - Return -1 if the input file is invalid, i.e. doesn't contain an LLVM
```

Comment:
Is there a test for this case(-1)?

Reply:
There isn't. Should I add the test cases for the error codes? I don't think it's something that's done for other tools, but doing so wouldn't be any trouble.

Reply:
Considering we are getting into the area of caring about the return codes, I think it's a good idea to add something.  Thanks

Reply:
Since negative numbers as error codes yield different things on Linux and Windows, I guess I'd better leave the former error codes undisturbed and test the '1' case for the actual "no SPIR kernel error".

Reply:
I do not see why we are checking `1` return code, when there is no other variants now...

Reply:
Covering the return code by the LIT feels right if the error code is documented in the code and relied upon by #1077. If, upon this basis, @mdtoguchi's request persists after my "no variant" changes, I would prefer to keep the checks.

Reply:
@mdtoguchi, do you know if there's a way to check the error code other than by `echo $?` within the LIT infrastructure?
Currently, the `$?` combination doesn't yield the last return code on Windows. Neither did the Windows-specific substitution via `%errorcode%` work. If there isn't a way to achieve this, there are basically 2 options:
1. Factor out the return code test into a separate file and mark it unsupported for Windows;
1. Drop the return code test altogether.

Reply:
I'm assuming you meant `%errorlevel%` and not `%errorcode%` here.  One thing I read was that `%errorlevel%` is not set when dealing with a windowed application run from the command prompt.  I don't know how llvm-lit invokes the commands so it may fall into that category for Windows.  The suggestion was to kick off the invocation with `start /wait <command>` which should set the `%errorlevel%`.  Maybe that would work.  All speculation of course.

We should at least have a test of some sort if we are relying on a different return code.  factoring out and marking as unsupported for Windows works for me.

Reply:
Yep, I've misspelled `%errorlevel%` for the comment.
Thanks! I'll experiment with these suggestions, resorting to the Windows-unsupported approach if nothing else comes.

Reply:
I've been oblivious to a simpler solution via Bash condition checking. Let's see if this ducks the issue for Windows.